text/0000-decorator-support.md

```diff
@@ -0,0 +1,142 @@
+- Start Date: 2019-02-06
+- Relevant Team(s): Ember.js, Ember Data, Ember CLI, Learning, Steering
+- RFC PR: https://github.com/emberjs/rfcs/pull/440
+- Tracking: (leave this empty)
+
+# Decorator Support
+
+## Summary
+
+Move forward with the Decorators RFC, and support decorators while they remain
+in stage 2 and move forward through the TC39 process.
+
+## Motivation
+
+The recently merged [Decorators
+RFC](https://github.com/emberjs/rfcs/blob/master/text/0408-decorators.md)
+contained an explicit condition: That it was premised on decorators moving from
+stage 2 in the TC39 process to stage 3. If this were to not happen, the original
+RFC stipulates that a followup (this RFC) should be created to clarify whether
+or not we should move forward with decorators while they remain in stage 2, and
+how we would do so.
+
+This RFC proposes that we do so. As discussed in the original, the decorator
+pattern has been in use in Ember since the beginning, and there is no easy path
+forward to native class syntax without a native equivalent to "classic"
+decorators.
+
+[Stage 2 in TC39](https://tc39.github.io/process-document/) signals that the
+committee expects this feature to be included in the language in the future, and
+is working out the details and semantics. The fact that decorators _remain_ in
+stage 2, and have not been rejected, signals that they still expect this to be
+the case. This is also the feedback that the core team received in the January
+TC39 meeting. _Crucially_, all parties were in agreement about the _invocation_
+syntax of decorators:
+
+```js
+class Person {
+  @decorate name;
+}
+```
+
+This is the most stable part of the spec, and the one that average developers
+will come into contact with most. It means that changes to the spec will likely
```

Comment:
Stage 2 settled on _decorators after _, but this could change since this has been a point of contention. However, no decorators in Ember itself are currently class decorators, and there aren't any plans currently to add them. Decorators in external libraries such `ember-decorators` will experience some amount of churn, but this should be very easy to codemod so it shouldn't have a huge impact.